dnfdaemon-server/package.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright (C) 2021 Red Hat, Inc.
+
+This file is part of dnfdaemon-server: https://github.com/rpm-software-management/libdnf/
+
+Dnfdaemon-server is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Dnfdaemon-server is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with dnfdaemon-server.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "package.hpp"
+
+#include <fmt/format.h>
+
+
+dnfdaemon::KeyValueMap package_to_map(
+    const libdnf::rpm::Package & libdnf_package, std::vector<std::string> & attributes) {
+    dnfdaemon::KeyValueMap dbus_package;
+    // add package id by default
+    dbus_package.emplace(std::make_pair("id", libdnf_package.get_id().id));
+    // attributes required by client
+    for (auto & attr : attributes) {
+        if (package_attributes.count(attr) == 0) {
+            throw std::runtime_error(fmt::format("Package attribute '{}' not supported", attr));
+        }
+        switch (package_attributes.at(attr)) {
```

Comment:
```
if (package_attributes.count(attr) == 0) {
    throw std::runtime_error(fmt::format("Package attribute '{}' not supported", attr));
}
switch (package_attributes.at(attr)) {
```
Why are you looking for an `attr` twice?
The `count` method looks for` attr` to count it.
If `attr` is found, then the `at` method looks for `attr` again (with an internal test for its existence).

How about searching for `attr` only once and if it is found, use the result (iterator)?
Eg.
```
auto it = package_attributes.find(attr);
if (it == package_attributes.end()) {
    throw std::runtime_error(fmt::format("Package attribute '{}' not supported", attr));
}
switch (it->second) {
```